pipeline/matcher_merger/merger/src/main/scala/weco/pipeline/merger/services/MergerWorkerService.scala

```diff
@@ -25,8 +29,10 @@ class MergerWorkerService[WorkDestination, ImageDestination](
   msgStream: SQSStream[NotificationMessage],
   sourceWorkLookup: IdentifiedWorkLookup,
   mergerManager: MergerManager,
-  workOrImageIndexer: Indexer[Either[Work[Merged], Image[Initial]]],
-  workMsgSender: MessageSender[WorkDestination],
+  workOrImageIndexer: Indexer[
+    Either[Either[Work[Merged], Work[Denormalised]], Image[Initial]]
```

Comment:
A mythical EitherEither, rarely spotted in the wild 👹 .

Can we express this without the nested Either?

Reply:
I've had a go at refactoring the `EitherIndexer` which underlies this issue and I can't find a way that is straightforward, without making a bigger change, so I think it makes sense to leave this for a refactor following the release of the merged merger/router 

I do think it could do with a refactor as the logic is quite convoluted, but lets not put it in the way of releasing this change.